### ISSUE TITLE
SpectralResidual results in an odd plot with instance_scores spiking only at the beginning and at the end and flattening at everywhere intermediate.

### ISSUE THREAD

**Issue by rajdeep-biswas** [SeldonIO/alibi-detect#290]
This is from a proprietary dataset with ~550k rows that I can not share here but here are the parameters -
```
od = SpectralResidual(
    threshold=1.,
    window_amp=20,
    window_local=20,
    n_est_points=10,
    n_grad_points=5
)
```

This results in the following (screenshot) -

![image](https://user-images.githubusercontent.com/32306614/125512854-29731a6d-6977-4755-8cd9-a357dbdec570.png)
```
'instance_score': array([0.52549029, 0.74446383, 0.65938995, ..., 0.54188361, 0.67161538,
         0.82524479])
```

Another example -

```
od = SpectralResidual(
    threshold=0.01,
    window_amp=5,
    window_local=5,
    n_est_points=2,
    n_grad_points=5
)
```

This results in -

![image](https://user-images.githubusercontent.com/32306614/125513262-2f793e56-c873-4991-8a11-cad3a5d43b58.png)
```
'instance_score': array([ 2.74739451, -0.1974772 , -0.65613109, ..., -0.04469103,
          0.27587623,  0.91586209])
```

**Comment by tnpradeep** [comment]
Hi,

Even I have observed similar behavior, whatever configuration you provide, always see spikes in beginning and end.
![image](https://user-images.githubusercontent.com/13568762/126621826-8d5e35d5-6987-4b8e-9194-b046abe2dc43.png)

Please explain why ? and how to make it better ?

Thanks
Pradeep

**Comment by RobertSamoilescu** [comment]
Spikes fixes in PR #396

**Comment by RobertSamoilescu** [comment]
@rajdeep-biswas, @tnpradeep, thank you for reporting the issue.
The PR #396 that fixed the issue has been merged into master.
